[2023-10-23T17:48:28.616+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:48:16.286339+00:00 [queued]>
[2023-10-23T17:48:28.631+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:48:16.286339+00:00 [queued]>
[2023-10-23T17:48:28.631+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2023-10-23T17:48:28.643+0000] {taskinstance.py:1382} INFO - Executing <Task(PostgresOperator): sat_employee_job_details_update> on 2023-10-23 17:48:16.286339+00:00
[2023-10-23T17:48:28.649+0000] {standard_task_runner.py:57} INFO - Started process 4484 to run task
[2023-10-23T17:48:28.653+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'update_dwh', 'sat_employee_job_details_update', 'manual__2023-10-23T17:48:16.286339+00:00', '--job-id', '243', '--raw', '--subdir', 'DAGS_FOLDER/update_dwh_dag.py', '--cfg-path', '/tmp/tmpx4hung79']
[2023-10-23T17:48:28.658+0000] {standard_task_runner.py:85} INFO - Job 243: Subtask sat_employee_job_details_update
[2023-10-23T17:48:28.716+0000] {task_command.py:416} INFO - Running <TaskInstance: update_dwh.sat_employee_job_details_update manual__2023-10-23T17:48:16.286339+00:00 [running]> on host 1f8bb8a692af
[2023-10-23T17:48:28.819+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='update_dwh' AIRFLOW_CTX_TASK_ID='sat_employee_job_details_update' AIRFLOW_CTX_EXECUTION_DATE='2023-10-23T17:48:16.286339+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-10-23T17:48:16.286339+00:00'
[2023-10-23T17:48:28.821+0000] {sql.py:274} INFO - Executing: 
    INSERT INTO TABLE rdv.sat_employee_job_details
    SELECT DISTINCT
        md5(so.employeenumber::text),
        'csv',
        now(),
        case when so.attrition = 'Yes' then true else false end,
        so.hourlyrate,
        so.joblevel,
        so.standardhours,
        so.stockoptionlevel
    FROM
        staging.stg_table so
    LEFT OUTER JOIN rdv.sat_employee_job_details sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
    where 
        coalesce(case when so.attrition = 'Yes' then true else false end, false) != coalesce(sat.attrition, false) or
        coalesce(so.joblevel, 0) != coalesce(sat.job_level, 0) or
        coalesce(so.hourlyrate, 0) != coalesce(sat.hourly_rate, 0) or
        coalesce(so.standardhours, 0) != coalesce(sat.standard_hours, 0) or
        coalesce(so.stockoptionlevel, 0) != coalesce(sat.stock_option_level, 0)
[2023-10-23T17:48:28.833+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-10-23T17:48:28.836+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/models/connection.py:374: AirflowProviderDeprecationWarning: The "schema" arg has been renamed to "database" as it contained the database name.Please use "database" to set the database name.
  return hook_class(**{hook.connection_id_attribute_name: self.conn_id}, **hook_params)

[2023-10-23T17:48:28.841+0000] {base.py:73} INFO - Using connection ID 'stg_conn' for task execution.
[2023-10-23T17:48:28.915+0000] {sql.py:418} INFO - Running statement: 
    INSERT INTO TABLE rdv.sat_employee_job_details
    SELECT DISTINCT
        md5(so.employeenumber::text),
        'csv',
        now(),
        case when so.attrition = 'Yes' then true else false end,
        so.hourlyrate,
        so.joblevel,
        so.standardhours,
        so.stockoptionlevel
    FROM
        staging.stg_table so
    LEFT OUTER JOIN rdv.sat_employee_job_details sat ON (sat.employee_hash_key = md5(so.employeenumber::text) AND sat.created_at IS NULL)
    where 
        coalesce(case when so.attrition = 'Yes' then true else false end, false) != coalesce(sat.attrition, false) or
        coalesce(so.joblevel, 0) != coalesce(sat.job_level, 0) or
        coalesce(so.hourlyrate, 0) != coalesce(sat.hourly_rate, 0) or
        coalesce(so.standardhours, 0) != coalesce(sat.standard_hours, 0) or
        coalesce(so.stockoptionlevel, 0) != coalesce(sat.stock_option_level, 0), parameters: None
[2023-10-23T17:48:28.936+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/operators/sql.py", line 280, in execute
    output = hook.run(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 423, in _run_command
    cur.execute(sql_statement)
psycopg2.errors.SyntaxError: syntax error at or near "TABLE"
LINE 2:     INSERT INTO TABLE rdv.sat_employee_job_details
                        ^

[2023-10-23T17:48:28.942+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=update_dwh, task_id=sat_employee_job_details_update, execution_date=20231023T174816, start_date=20231023T174828, end_date=20231023T174828
[2023-10-23T17:48:28.959+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 243 for task sat_employee_job_details_update (syntax error at or near "TABLE"
LINE 2:     INSERT INTO TABLE rdv.sat_employee_job_details
                        ^
; 4484)
[2023-10-23T17:48:28.990+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-23T17:48:29.004+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
